components/scream/src/share/util/scream_time_stamp.cpp

```diff
@@ -94,6 +94,10 @@ int TimeStamp::seconds_from (const TimeStamp& ts) const {
   return *this-ts;
 }
 
+double TimeStamp::days_from (const TimeStamp& ts) const {
+  return ((double)*this.seconds_from(ts)/86400.0);
```

Comment:
Fun fact: if one of the two operands is a double, the other is automatically promoted to double. So this works just as well:
```
  return seconds_from(ts) / 86400.0;
```
since the denominator is double.

Reply:
should I change it?  Or just stick with what's there, since both work the same way?

Reply:
I don't understand the current implementation. Is this taking this.seconds_from(ts) and multiplying it by double to promote it to a double from an int, then dividing it by a double which would - as Luca notes - have an identical effect? If so, then yeah - it would probably be better to just divide by a double to get the type you want.

I'm also a bit confused why we create a new function `days_from` which just divides `seconds_from` by 86400.0... which is exactly what I had already done in the code directly. I'm not a fan of creating functions which just create extra steps and obscure what we're doing, so I'd rather keep the code as I originally implemented it unless I'm missing something.

Reply:
That's alright with me.  We can just revert the branch to the commit before this one.  I put in the implementation based on this comment from above,
"
Yeah, I agree that days_from would be better than seconds_from. Would you mind making that change, Aaron? I couldn't find where seconds_from was defined by grepping or I would have made that change myself...
"

Reply:
Peter, the bottom line is code clarity. Reading `ts.seconds_from(t0) / 86400.0` requires (a tiny bit) more mental parsing than `ts.days_from(t0)`. It communicates "what" you want, hiding a bit "how" you get it.

I'm not sure if it's a personal taste though. Since it's used only in one place, and it's a quite simple and obvious calculation (we all know there are 86400 seconds in a day), it might be not that important.

Reply:
Yeah, sorry for asking you to do extra work. I had assumed that you had created `seconds_from` yourself and were going to create a `days_from` routine as an equivalent low-level function. Since using `seconds_from` then dividing by 86400 is kind of a kludge, I figured it would be better to instead create a function that returns days. But if you're just going to use the same kludge as I had, there's not point!

Reply:
Yeah, ok Luca. I don't care which method we use. If either of you have a preference, let's go with that. Maybe "what we already have" is best? In which case, I think if one of you approve the code will auto-merge?

Reply:
yeah, this is a tiny detail. I don't really care either. :)